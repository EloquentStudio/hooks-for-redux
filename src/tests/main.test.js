// index.js
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import { getStore } from '../index'

import {subscribeToName, updateName} from './NameHook.js'

const App = () =>
  <p onClick={
    () => updateName((name) => name == 'Alice' ? 'Bob' : 'Alice')
  }>
    Hello there, {subscribeToName()}! Click to change me.
  </p>

it('renders without crashing', () => {
  // const div = document.createElement('div');
  const component = renderer.create(
    <Provider store={getStore()}>
    <App />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  renderer.act(() => (updateName("Shane"),undefined))
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // ReactDOM.unmountComponentAtNode(div);
});
