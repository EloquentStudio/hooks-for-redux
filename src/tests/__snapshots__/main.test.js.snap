// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<p
  onClick={[Function]}
>
  Hello there, 
  Alice
  ! Click to change me.
</p>
`;

exports[`renders without crashing 2`] = `
<p
  onClick={[Function]}
>
  Hello there, 
  Shane
  ! Click to change me.
</p>
`;
